source/server-discovery-and-monitoring/tests/unified/logging-standalone.json

```diff
@@ -0,0 +1,520 @@
+                "closeConnection": true
+              }
+            }
+          }
+        },
+        {
+          "name": "waitForEvent",
+          "object": "testRunner",
+          "arguments": {
+            "client": "client",
+            "event": {
+              "serverHeartbeatFailedEvent": {}
+            },
+            "count": 1
+          }
+        }
+      ],
+      "expectLogMessages": [
+        {
+          "client": "client",
+          "ignoreExtraMessages": true,
+          "ignoreMessages": [
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Topology description changed"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Stopped topology monitoring"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Stopped server monitoring"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Topology description changed"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Starting server monitoring"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Starting topology monitoring"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Server heartbeat started"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Server heartbeat succeeded"
+              }
+            }
+          ],
+          "messages": [
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Server heartbeat failed",
+                "awaited": {
+                  "$$exists": true
+                },
+                "topologyId": {
+                  "$$exists": true
+                },
+                "serverHost": {
+                  "$$type": "string"
+                },
+                "serverPort": {
+                  "$$type": [
+                    "int",
+                    "long"
+                  ]
+                },
+                "driverConnectionId": {
+                  "$$exists": true
+                },
+                "serverConnectionId": {
+                  "$$exists": true
+                },
```

Comment:
This check (and ditto in the "Failing heartbeat" tests for other topologies) is flaky for me in CI. I *think* it's due to a race between setting the failpoint for hellos and establishing a monitoring connection. If a monitor establishes a connection first, the sequence of events will look something like this:

1. Send a hello for initial connection establishment. Emit a **"Server heartbeat started"** log message with no serverConnectionId, as we haven't talked to the server yet.
2. The server replies with a successful hello response, including a serverConnectionId. Emit a **"Server heartbeat succeeded"** message containing the serverConnectionId from the response.
3. The failpoint for hellos is set.
4. Begin a new server heartbeat and emit a **"Server heartbeat started"** log message with the serverConnectionId stored from the previous successful response.
5. The heartbeat fails. Emit a **"Server heartbeat failed"** log message with the serverConnectionId stored from the previous successful response.

However, if the failpoint is set before the monitoring connection is established, the following will occur:
1. The failpoint for hellos is set.
2. Send a hello for initial connection establishment. Emit a **"Server heartbeat started"** log message with no serverConnectionId, as we haven't talked to the server yet.
3. The heartbeat fails. Emit a **"Server heartbeat failed"** log message. The serverConnectionId will not be present because we do not have a stored value from a previous successful heartbeat.

Let me know if this makes sense to you. If so, I think we'll need to remove this check from each "Failing heartbeat" test because of the previously discussed discrepancy between drivers regarding when heartbeat events begin to be emitted.

Reply:
Yes this makes sense, thanks for catching this @isabelatkinson.
Could you please confirm that tests are stable after removing `serverConnectionId`?

Reply:
Yes, I get a clean patch with the ID removed from all of the failing heartbeat tests.